### DISCUSSION TITLE
not found: npm on zowe plugins install

### DISCUSSION THREAD

**Discussion by dkelosky** [zowe/zowe-cli#1613] [Q&A] [answered]
Hi - a user is getting this error "not found: npm" when running `zowe plugins install ...`; however, as you can see below, npm version is 9.3.1.  He's using zowe 7.9.7 

```
C:\>zowe plugins install @broadcom/ops-for-zowe-cli
Handler Instantiation Failed:
Could not instantiate the handler C:\Users\gl891170\AppData\Roaming\npm\node_modules\@zowe\cli\node_modules\@zowe\imperative\lib\imperative\src\plugins\cmd\install\install.handler for command install
Error Details:
not found: npm

C:\>npm --version
9.3.1
```

what do

**Comment by t1m0thyj** [comment]
@dkelosky Could you have the user run the following commands and share the output?
```
npx -y which "npm"
npx -y which "npm.cmd"
```

The first command may fail, since "which" is the same package that Imperative is using to look for npm on the PATH.

Curious if the second command succeeds, perhaps the failure is due to the ".cmd" suffix being required in some Windows environments?

**Comment by dkelosky** [comment]
![image](https://user-images.githubusercontent.com/14940561/215119103-e781ea0f-bb12-4694-96b0-44bc8ff83a8e.png)

**Comment by zFernand0** [comment]
The resolution seem to have been an issue (#1615) and it should be fixed in `@zowe/imperative@5.8.2`
Credits to @traeok, @awharn, and @t1m0thyj.

**Reply by dkelosky** [reply]
imo we should also credit the people who pester @t1m0thyj , @awharn , @traeok , and @zFernand0 for a fix.

**Comment by dkelosky** [comment, marked as answer]
q: what do
a: upgrade zowe cli
